src/test/java/org/apache/commons/lang3/time/StopWatchTest.java

```diff
@@ -448,4 +470,712 @@ public void testToStringWithMessage() throws InterruptedException {
+            final StopWatch watch = StopWatch.create();
+
+            watch.start();
+            watch.suspend();
+
+            assertTrue(watch.isSuspended());
+
+            BiPredicate<String, String> predicate = Objects::equals;
+
+            long result = Streams.of(ImmutablePair.of("A", "A"), ImmutablePair.of("A", "B"))
+                                 .filter(it -> watch.test(predicate).test(it.getLeft(), it.getRight()))
+                                 .count();
+
+            assertEquals(1, result);
+            assertTrue(watch.isSuspended());
+        }
+
+        @Test
+        void testTestHandlesExceptionsProperly() {
+            final StopWatch watch = StopWatch.create();
+
+            assertTrue(watch.isStopped());
+
+            BiPredicate<String, String> predicate = (a, b) ->  {
+                throw new RuntimeException();
+            };
+
+            assertThrows(RuntimeException.class, () -> watch.test(predicate).test("a", "b"));
+            assertTrue(watch.isSuspended());
+        }
+    }
+
+    @Nested
+    public class FailableBiPredicateSupportRelatedTests {
+        @Test
+        public void testTest() {
+            final StopWatch watch = StopWatch.create();
+
+            FailableBiPredicate<String, String, IllegalArgumentException> predicate = Objects::equals;
+
+            long result = Streams.of(ImmutablePair.of("A", "A"), ImmutablePair.of("A", "B"))
+                                 .filter(it -> watch.test(predicate).test(it.getLeft(), it.getRight()))
+                                 .count();
+
+            assertEquals(1, result);
+            assertTrue(watch.isSuspended());
+        }
+
+        @Test
+        public void testTestWhenStopWatchHasBeenSuspended() {
+            final StopWatch watch = StopWatch.create();
+
+            watch.start();
+            watch.suspend();
+
+            assertTrue(watch.isSuspended());
+
+            FailableBiPredicate<String, String, IllegalArgumentException> predicate = Objects::equals;
+
+            long result = Streams.of(ImmutablePair.of("A", "A"), ImmutablePair.of("A", "B"))
+                                 .filter(it -> watch.test(predicate).test(it.getLeft(), it.getRight()))
+                                 .count();
+
+            assertEquals(1, result);
+            assertTrue(watch.isSuspended());
+        }
+
+        @Test
+        public void testTestHandlesExceptionsProperly() {
+            final StopWatch watch = StopWatch.create();
+
+            FailableBiPredicate<String, String, IllegalArgumentException> predicate = (a, b) -> {
+                throw new IllegalArgumentException();
+            };
+
+            assertThrows(IllegalArgumentException.class, () -> watch.test(predicate).test("a", "b"));
+
+            assertTrue(watch.isSuspended());
+        }
+    }
+
+    @Nested
+    public class FailableBiConsumerSupportRelatedTests {
+        @SuppressWarnings("unchecked")
```

Comment:
Remove all `@SuppressWarnings("unchecked")`s and use matching a method matching the class, for example `FailableBiConsumer.nop()`.

Reply:
Removed annotations